config/puma.rb

```diff
@@ -1,28 +1,43 @@
-# https://devcenter.heroku.com/articles/deploying-rails-applications-with-the-puma-web-server
+# Puma can serve each request in a thread from an internal thread pool.
+# The `threads` method setting takes two numbers: a minimum and maximum.
+# Any libraries that use thread pools should be configured to match
+# the maximum value specified for Puma. Default is set to 5 threads for minimum
+# and maximum; this matches the default thread size of Active Record.
+#
+max_threads_count = ENV.fetch("RAILS_MAX_THREADS") { 5 }
+min_threads_count = ENV.fetch("RAILS_MIN_THREADS") { max_threads_count }
+threads min_threads_count, max_threads_count
+
+# Specifies the `worker_timeout` threshold that Puma will use to wait before
+# terminating a worker in development environments.
+#
+worker_timeout 3600 if ENV.fetch("RAILS_ENV", "development") == "development"
 
-# The environment variable WEB_CONCURRENCY may be set to a default value based
-# on dyno size. To manually configure this value use heroku config:set
-# WEB_CONCURRENCY.
+# Specifies the `port` that Puma will listen on to receive requests; default is 3000.
 #
-# Increasing the number of workers will increase the amount of resting memory
-# your dynos use. Increasing the number of threads will increase the amount of
-# potential bloat added to your dynos when they are responding to heavy
-# requests.
+port ENV.fetch("PORT") { 3000 }
+
+# Specifies the `environment` that Puma will run in.
 #
-# Starting with a low number of workers and threads provides adequate
-# performance for most applications, even under load, while maintaining a low
-# risk of overusing memory.
-workers Integer(ENV.fetch("WEB_CONCURRENCY", 2))
-threads_count = Integer(ENV.fetch("MAX_THREADS", 2))
-threads(threads_count, threads_count)
```

Comment:
Can you double-check this configuration file?

Reply:
Absolutely!

Reply:
```
workers Integer(ENV.fetch("WEB_CONCURRENCY", 2))
```

is now
```
workers ENV.fetch("WEB_CONCURRENCY") { 2 }
```

I did not uncommenting this earlier but it's fixed now.

Reply:
```
threads_count = Integer(ENV.fetch("MAX_THREADS", 2))
threads(threads_count, threads_count)
```

is now
```
max_threads_count = ENV.fetch("RAILS_MAX_THREADS") { 5 }
min_threads_count = ENV.fetch("RAILS_MIN_THREADS") { max_threads_count }
```

at the top of this configuration file.

What else did I miss? 🙂

Reply:
We must check if we have set the `MAX_THREADS` environment variable in any of our environments. The new configuration will ignore that value. You want to change the `max_threads_count` definition to the following. What do you think?

```ruby
max_threads_count = ENV.fetch("MAX_THREADS") { 2 }
```

Reply:
Good call! We are currently not setting `MAX_THREADS` in any of our environments.

Reply:
We are currently not setting `MAX_THREADS` in any of our environments.